eos_data_distribution/avahi_monitor.py

```diff
@@ -188,6 +192,12 @@ def remove_nexthop(self, faceUri):
         return check_call(["nfdc", "remove-nexthop",
                            str(SUBSCRIPTIONS_SOMA), faceUri])
 
+    def start_edge(self):
+        return check_call(["systemctl", "start", "edd-soma-subscriptions-producer"])
```

Comment:
This is shipping on EOS: it will always have Avahi, right? In any case, the code here is essentially a policy for when to enable the SOMA producer. If we move this code into `edd-soma-subscriptions-producer` there is nothing stopping it also having other policies which could override this. Probably via a config file.